odl/solvers/util/callback.py

```diff
@@ -854,6 +856,52 @@ def __repr__(self):
             self.logy)
 
 
+class CallbackPrintMemory(SolverCallback):
+
+    """Callback for printing memory and CPU usage."""
+
+    def __init__(self, step=1):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        step : positive int, optional
+            Number of iterations between output. Default: 1
+
+        Examples
+        --------
+        Print memory and CPU usage
+
+        >>> callback = CallbackPrintMemory()
+
+        Only print every tenth step
+
+        >>> callback = CallbackPrintMemory(step=10)
+        """
+        self.step = step
+        self.iter = 0
+
+    def __call__(self, _):
+        """Print the memory and CPU usage"""
+        if self.iter % self.step == 0:
+            print('CPU usage (% each core): {}'.format(
```

Comment:
Suggestion:
- remove swap (not really needed)
- use format strings for each resource separately, like `fmt_cpu`, `fmt_mem`, with sensible defaults
- add flags `print_cpu` and `print_mem` to switch on/off